docs/datamodel/links.rst

```diff
@@ -14,16 +14,259 @@ There are two kinds of link item declarations: *abstract links*, and
-types.
-
-Links are directional and have a *source* and one or more *targets*.
-The number of targets is specified by the keywords :ref:`required
-<ref_eql_ddl_links_syntax>`, :ref:`single <ref_eql_ddl_links_syntax>`,
-and :ref:`multi <ref_eql_ddl_links_syntax>`.  It is also possible to
-restrict how many source objects can link to the same target via the
-:eql:constraint:`exclusive` constraint.
+:ref:`link properties <ref_datamodel_props>`, or setup
+:ref:`constraints <ref_datamodel_constraints>`.  Concrete links
+are defined on specific object types.
+
+Links are directional and have a *source*. Whether a *source* has one
+or more links of the same kind is specified by the keywords
+:ref:`single <ref_eql_ddl_links_syntax>` and :ref:`multi
+<ref_eql_ddl_links_syntax>`, respectively. The :ref:`required
+<ref_eql_ddl_links_syntax>` keyword indicates that at least one
+target object must be linked for a particular link kind. It is
+also possible to restrict how many source objects can link to the
+same target via the :eql:constraint:`exclusive` constraint. Using
+these tools it's possible to specify common relationships between
+things: *many-to-one*, *one-to-one*, and *many-to-many*.
+
+
+Many-to-One
+-----------
+
+A *many-to-one* relationship is a fairly common pattern representing
+situations like ownership or hierarchies. For example, ``Person`` and
+``Shirt``:
+
+.. code-block:: sdl
+
+    type Person {
+        required property name -> str {
+            constraint exclusive;
+        }
+    }
+    type Shirt {
+        required property description -> str {
+            # Just making sure that each description
+            # is unique like a name.
+            constraint exclusive;
+        }
+        link owner -> Person;
+    }
+
+A ``Shirt`` can have at most one ``owner``, while a ``Person`` can
+have potentially have more than one ``Shirt``. This is a *many-to-one*
+relationship and it's expressed by the ``link owner``.
+
+Selecting the shirts belonging to a specific owner can be done with
+the following query:
+
+.. code-block:: edgeql
+
+    SELECT Shirt {
+        description,
+        owner: {
+            name
+        }
+    }
+    FILTER .owner.name = 'Billie';
+
+It is possible to traverse any link backwards. When a *many-to-one*
+link is traversed backwards, the result represents a *one-to-many*
+relationship instead. For example, the previous query can be
+re-written like this:
+
+.. code-block:: edgeql
+
+    SELECT Person {
+        name,
+        # let's use a computable here
+        shirts := .<owner[IS Shirt] {
+            description
+        }
+    }
+    FILTER .name = 'Billie';
+
+Alternatively, the above relationship can also be represented by the
+following schema:
+
+.. code-block:: sdl
+
+    type Person {
+        required property name -> str {
+            constraint exclusive;
+        }
+        multi link shirts -> Shirt {
+            constraint exclusive;
```

Comment:
I would put some stress on `exclusive` here. This is what makes `many-to-one` instead of `many-to-many` in this case. Otherwise it's easy to miss when making your own schema.

Reply:
I've added a comment to the schema.